neo/SmartContract/NeoService.cs

```diff
@@ -165,6 +169,36 @@ public NeoService(TriggerType trigger, Snapshot snapshot)
             #endregion
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="engine">r, s, v, messageHash</param>
+        /// <returns></returns>
+        private bool Ecrecover(ExecutionEngine engine)
+        {
+            var r = new System.Numerics.BigInteger(engine.CurrentContext.EvaluationStack.Pop().GetByteArray().Reverse().Concat(new byte[1]).ToArray());
+            var s = new System.Numerics.BigInteger(engine.CurrentContext.EvaluationStack.Pop().GetByteArray().Reverse().Concat(new byte[1]).ToArray());
+            bool v = engine.CurrentContext.EvaluationStack.Pop().GetBoolean();
+            byte[] messageHash = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            try
+            {
+                ECPoint point = ECDsa.KeyRecover(ECCurve.Secp256k1, r, s, messageHash, v, true);
```

Comment:
The curve should be other param, or create different syscalls for different curves.